src/pico_ssd1306/CMakeLists.txt

```diff
@@ -0,0 +1,26 @@
+cmake_minimum_required(VERSION 3.12)
```

Comment:
after this point is mostly screen library code. I don't want to get into my modifications, as most of them were just allowing me to import the code as libraries, or adapting them to my specific display. This code should be in `external/` - and some of my changes might not be perfectly optimised.